File: src/coreclr/tools/aot/jitinterface/jitinterface.cpp
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

#include <stdarg.h>
#include <stdlib.h>
#include <stdint.h>

#include "dllexport.h"
#include "jitinterface_generated.h"

static void NotImplemented()
{
    abort();
}

bool JitInterfaceWrapper::runWithErrorTrap(ICorJitInfo::errorTrapFunction function, void* parameter)
{
    try
    {
        (*function)(parameter);
    }
    catch (CorInfoExceptionClass *)
    {
        return false;
    }
    return true;
}

bool JitInterfaceWrapper::runWithSPMIErrorTrap(ICorJitInfo::errorTrapFunction function, void* parameter)
{
    try
    {
        (*function)(parameter);
    }
    catch (CorInfoExceptionClass *)
    {
        return false;
    }
    return true;
}

File: src/coreclr/tools/aot/ILCompiler.Compiler.Tests/ILCompiler.Compiler.Tests.Assets/JitInterface.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace ILCompiler.Compiler.Tests.Assets
{
    public class JitInterface
    {
        public struct Struct
        {
            public object Reference;
        }

        public struct BlittableStruct
        {
            public int Value;
        }

        public int Instance;
        public static int Primitive;
        public static object Reference;
        public static BlittableStruct StructWithoutReference;
        public static Struct StructWithReference;
    }
}
